databases/week1/queries.sql

```diff
@@ -0,0 +1,32 @@
+-- 1. Find out how many tasks are in the task table
+SELECT count(title) FROM task;
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(title) FROM task WHERE due_date IS NULL;
+-- 3.Find all the tasks that are marked as done
+SELECT * FROM task
+JOIN status ON task.status_id = status.id 
+WHERE status.name = 'Done';
+--4. Find all the tasks that are not marked as done
+SELECT * FROM task
+JOIN status ON task.status_id = status.id
+WHERE status.name != 'Done';
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT * FROM task ORDER BY created DESC;
+--6.Get the single most recently created task
+SELECT * FROM task ORDER BY created DESC LIMIT 1;
+--7. Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date FROM task
+WHERE title LIKE '%database%' OR description LIKE '%database%';
+--8. Get the title and status (as text) of all tasks
+SELECT task.title, status.name FROM task, status WHERE task.status_id=status.id;
```

Comment:
I would use the join clause instead here as you are extracting data from two tables. https://www.w3schools.com/sql/sql_join.asp

Reply:
Thank you for the recommendation, I will keep it in my mind :)